sw/otbn/crypto/p256.s

```diff
@@ -1691,6 +1693,133 @@ p256_scalar_mult:
 
   ret
 
+/**
+ * Generate a nonzero random value in the scalar field.
+ *
+ * Returns t, a random value in the range [1,n-1].
+ *
+ * This follows the method in FIPS 186-4 sections B.4.2 and B.5.2 for
+ * generation of secret scalar values d and k. The computation is:
+ *   do {
+ *     c = RBG(256); // fetch 256b random value
+ *   } while (c >= n - 1)
+ *   return c + 1;
+ *
+ * This implementation handles the unmasked secret value, but the seed is
+ * pulled from RND so it cannot be re-run with the same seed. This method
+ * should not be used for keymgr-derived seeds! However, it masks the secret
```

Comment:
Do we have any way to "promote" this caveat/warning so that no one accidentally does this?  I sort of want the equivalent of bold and 32-point-font on it :)

If not, it's documented clearly here, but foot-shooting should be discouraged as best we can.

Reply:
I agree but I think there are some built-in defenses here against foot-shooting. It would be pretty nontrivial to adapt this implementation to use a keymgr seed, since I used the method that requires repeated generations of random seeds; this would require repeated calls to keymgr, which would in practice mean jumping back and forth between Ibex and OTBN. Alternatively someone could implement the alternative, single-seed algorithm, which looks a bit different. But either way, they'd need to extensively modify the code to fall into this trap (and hopefully they'd come across the comment and think twice in the course of doing that). I also plan to add another variant that uses a keymgr seed, once I figure out how to do it safely, which should be another clue to think twice.